### PR TITLE
attempted to add coin functionality

### PR DESCRIPTION
**Pull request by dstrizzi**
dude...what am I even doing.

Successes:
-Added coin functionality. 
-Successfully adds coins to the total. 
-Undoing a selection reverts back to original board**

Todos:
-Add coin bonuses when players reach 4, 8, and 12 coins

**Undo-ing an option breaks when replaced with a new card

### CODE REVIEW

File: src/cards.js [hunk 1]
```diff
@@ -117,7 +137,13 @@ export const TREASURES = [
     value: 12,
     badge: false,
     grid: [
+      [33, 0, 1, 0],
       [0, 0, 1, 0],
+      [0, 0, 0, 0],
+      [1, 1, 0, 0],
+    ],
+    originalGrid: [
+      [33, 0, 1, 0],
```
Comment (bmorelli25) [inline src/cards.js]: why 33?

File: src/cards.js [hunk 2]
```diff
@@ -117,7 +137,13 @@ export const TREASURES = [
     value: 12,
     badge: false,
     grid: [
+      [33, 0, 1, 0],
```
Comment (bmorelli25) [inline src/cards.js]: same here, 33

File: src/cards.js [hunk 3]
```diff
@@ -81,31 +81,51 @@ export const TREASURES = [
...
     badge: false,
     grid: [
-      [0, 0, 1, 1],
+      [33, 0, 1, 1],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [1, 1, 0, 0],
+    ],
+    originalGrid: [
+      [33, 0, 1, 1],
```
Comment (bmorelli25) [inline src/cards.js]: 33

File: src/cards.js [hunk 4]
```diff
@@ -81,31 +81,51 @@ export const TREASURES = [
...
     bonus: null,
+    isCoin: true,
   },
   {
     color: "orange",
     value: 12,
     badge: false,
     grid: [
-      [0, 0, 1, 1],
+      [33, 0, 1, 1],
```
Comment (bmorelli25) [inline src/cards.js]: 33

File: src/cards.js [hunk 5]
```diff
@@ -81,31 +81,51 @@ export const TREASURES = [
...
-      [1, 1, 0, 0],
+      [1, 1, 0, 3],
       [0, 0, 0, 0],
       [0, 0, 0, 0],
       [0, 0, 0, 0],
     ],
+    originalGrid: [
+      [1, 1, 0, 3],
+      [0, 0, 0, 0],
+      (33)[(0, 0, 0, 0)],
```
Comment (bmorelli25) [inline src/cards.js]: 33

File: src/Square.js [hunk 6]
```diff
@@ -1,9 +1,14 @@
-import React from "react";
-import "./square.css"
+import React, { useState } from "react";
```
Comment (bmorelli25) [inline src/Square.js]: is useState used in this file?

File: src/cards.js (hunk 5 above)
Comment (dstrizzi) [reply src/cards.js]: these should all be 3's

File: src/Square.js (hunk 6 above)
Comment (dstrizzi) [reply src/Square.js]: No, but it almost was...

File: src/CardContext.js [hunk 7]
```diff
@@ -40,11 +40,13 @@ const CardContextProvider = (props) => {
...
-    playerOne: { current: [], complete: [], score },
-    playerTwo: { current: [], complete: [], score },
-    playerThree: { current: [], complete: [], score },
-    playerFour: { current: [], complete: [], score },
+    playerOne: { current: [], complete: [], score, coins: 0 },
+    playerTwo: { current: [], complete: [], score, coins: 0 },
+    playerThree: { current: [], complete: [], score, coins: 0 },
+    playerFour: { current: [], complete: [], score, coins: 0 },
     roundSubmits: [0, 0, 0, 0],
+    coins: [0, 0, 0, 0],
```
Comment (bmorelli25) [inline src/CardContext.js]: What's the difference between state.player*.coins (L43-L46) and the array in state.coins (L48)?

File: src/cards.js (hunk 5 above)
Comment (bmorelli25) [reply src/cards.js]:
```suggestion
      [3, 0, 0, 0],
```

File: src/cards.js (hunk 4 above)
Comment (bmorelli25) [inline src/cards.js]:
```suggestion
      [3, 0, 1, 1],
```

File: src/cards.js (hunk 3 above)
Comment (bmorelli25) [reply src/cards.js]:
```suggestion
      [3, 0, 1, 1],
```

File: src/cards.js (hunk 2 above)
Comment (bmorelli25) [reply src/cards.js]:
```suggestion
      [3, 0, 1, 0],
```

File: src/cards.js (hunk 1 above)
Comment (bmorelli25) [reply src/cards.js]:
```suggestion
      [3, 0, 1, 0],
```

File: src/Square.js (hunk 6 above)
Comment (bmorelli25) [reply src/Square.js]:
```suggestion
import React from "react";
```

File: src/CardContext.js (hunk 7 above)
Comment (dstrizzi) [reply src/CardContext.js]: state.player*coins can be removed. Early attempt

File: src/cards.js [hunk 8]
```diff
@@ -81,31 +81,51 @@ export const TREASURES = [
...
       [0, 0, 0, 0],
+    ],
+    originalGrid: [
+      [3, 0, 0, 0],
       [0, 1, 0, 0],
       [0, 0, 1, 0],
       [0, 0, 0, 0],
     ],
     bonus: null,
+    isCoin: true,
```
Comment (bmorelli25) [inline src/cards.js]:
```suggestion
```

File: src/cards.js [hunk 9]
```diff
@@ -81,31 +81,51 @@ export const TREASURES = [
...
       [0, 0, 0, 0],
     ],
+    originalGrid: [
+      [1, 1, 0, 3],
+      [0, 0, 0, 0],
+      (33)[(0, 0, 0, 0)],
+      [0, 0, 0, 0],
+    ],
     bonus: null,
+    isCoin: true,
```
Comment (bmorelli25) [inline src/cards.js]:
```suggestion
```

File: src/Board.js [hunk 10]
```diff
@@ -23,13 +25,14 @@ export default function Board({
   ];
 
   const [turn, setTurn] = useState(blankBoard);
+  const [hasCoin, setHasCoin] = useState(false);
```
Comment (bmorelli25) [inline src/Board.js]: A problem I anticipate is that there are cards and shapes where it's possible to get two coins with a single shape. Using a boolean here only allows one coin per shape

Comment (bmorelli25) [reply src/Board.js]: A counter would fix this

File: src/Board.js [hunk 11]
```diff
@@ -60,12 +74,28 @@ export default function Board({
...
+        }}
+      />
+    );
   }
 
+  //Every time coinUpdate updates (which is whenever a coin gets added), this fuction runs
+  useEffect(() => {
+    //add the coins to score when players reach 4, 8, and 12.
+    //Do this with state.coinBonus.pop() to assign the score and remove it from the array
+  }, [coinUpdate]);
```
Comment (bmorelli25) [inline src/Board.js]: This is neat. So this does nothing right now but will in the future?

File: src/CardContext.js [hunk 12]
```diff
@@ -40,11 +40,13 @@ const CardContextProvider = (props) => {
...
     currentPhase: 1, // four phases per game
     numPlayers: null,
-    playerOne: { current: [], complete: [], score },
-    playerTwo: { current: [], complete: [], score },
-    playerThree: { current: [], complete: [], score },
-    playerFour: { current: [], complete: [], score },
+    playerOne: { current: [], complete: [], score, coins: 0 },
+    playerTwo: { current: [], complete: [], score, coins: 0 },
+    playerThree: { current: [], complete: [], score, coins: 0 },
+    playerFour: { current: [], complete: [], score, coins: 0 },
```
Comment (bmorelli25) [inline src/CardContext.js]:
```suggestion
    playerOne: { current: [], complete: [], score, },
    playerTwo: { current: [], complete: [], score, },
    playerThree: { current: [], complete: [], score, },
    playerFour: { current: [], complete: [], score, },
```

**Review by bmorelli25 — APPROVED** [review]
We need to add logic for multiple coins in a single shape, and improve the style so you can still see the coin when selected, but by god, this is nice!

**Pull request merged**